src/GraphQL/ObjectExtensions.cs

```diff
@@ -296,25 +296,31 @@ public static T GetPropertyValue<T>(this object value)
         /// <param name="value">The value being tested.</param>
         public static bool IsDefinedEnumValue(Type type, object value)
         {
-            var names = Enum.GetNames(type);
-            if (names.Contains(value?.ToString() ?? "", StringComparer.OrdinalIgnoreCase))
+            try
             {
-                return true;
-            }
+                var names = Enum.GetNames(type);
+                if (names.Contains(value?.ToString() ?? "", StringComparer.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
 
-            var underlyingType = Enum.GetUnderlyingType(type);
-            var converted = Convert.ChangeType(value, underlyingType);
+                var underlyingType = Enum.GetUnderlyingType(type);
+                var converted = Convert.ChangeType(value, underlyingType);
 
-            var values = Enum.GetValues(type);
+                var values = Enum.GetValues(type);
 
-            foreach (var val in values)
-            {
-                var convertedVal = Convert.ChangeType(val, underlyingType);
```

Comment:
This line of code, which is questionable if it belongs here or not, throws an exception when called with an invalid enum string.